### PR TITLE
(fix) Strip trailing slash in api_base for prometheus metrics

### PR DESCRIPTION
**Pull request by owaisaamir**
### What change is being made?

Remove trailing slashes from `api_base` in the prometheus metrics

### Why are these changes being made?

Trailing slashes in `api_base` values can cause increased metric cardinality resulting in higher custom metrics cost in platform such as Datadog. This can also cause metrics not getting aggregated correctly for the same `api_base`.

example metrics with trailing slash:

```
litellm_deployment_success_responses_total{api_base="https://api.openai.com/v1/",api_key_alias="None",api_provider="openai",hashed_api_key="None",litellm_model_name="gpt-4o",model_id="0816ddaaddd6d762d4ba80f754895c0386360bdecc35430e5644bba7eb905b4f",requested_model="openai-gpt-4o",team="None",team_alias="None"} 4.0
litellm_deployment_success_responses_total{api_base="https://my-test-instance.openai.azure.com//openai/",api_key_alias="None",api_provider="azure",hashed_api_key="None",litellm_model_name="gpt-4o",model_id="4346bb17b8f946ed63b7a315d50b765be26387b0ffe3f1875c37a1efdcf0e82e",requested_model="azure-gpt-4o",team="None",team_alias="None"} 3.0
litellm_deployment_success_responses_total{api_base="https://api.openai.com/v1",api_key_alias="None",api_provider="openai",hashed_api_key="EMPTY",litellm_model_name="gpt-4o",model_id="0816ddaaddd6d762d4ba80f754895c0386360bdecc35430e5644bba7eb905b4f",requested_model="openai-gpt-4o",team="None",team_alias="None"} 1.0
```

By stripping trailing slashes, this improves metrics aggregation and reduces custom metrics cost.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: litellm/router_utils/cooldown_callbacks.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ async def router_cooldown_event_callback(
     _api_base = (
         litellm.get_api_base(model=_model_name, optional_params=temp_litellm_params)
         or ""
-    )
+    ).rstrip("/")
```
Comment (ghost) [inline litellm/router_utils/cooldown_callbacks.py:47]: why is this needed here? @owaisaamir

File: litellm/integrations/prometheus.py [hunk 2]
```diff
@@ -789,7 +789,7 @@ def set_llm_deployment_failure_metrics(self, request_kwargs: dict):
             _litellm_params = request_kwargs.get("litellm_params", {}) or {}
             litellm_model_name = request_kwargs.get("model", None)
             model_group = standard_logging_payload.get("model_group", None)
-            api_base = standard_logging_payload.get("api_base", None)
+            api_base = self._strip_trailing_slash(standard_logging_payload.get("api_base", None))
```
Comment (ghost) [inline litellm/integrations/prometheus.py:792]: this should just be done on the standard_logging_payload

File: litellm/router_utils/cooldown_callbacks.py (hunk 1 above)
Comment (owaisaamir) [reply litellm/router_utils/cooldown_callbacks.py:47]: This uses a different approach to get api_base compared to other prometheus metrics. It will be better to have a standard approach for all metrics.

File: litellm/integrations/prometheus.py (hunk 2 above)
Comment (owaisaamir) [reply litellm/integrations/prometheus.py:792]: Looks like you have refactored and pushed the change. Thanks for taking the fix.
